src/XrdOssStats/XrdOssStatsFileSystem.hh

```diff
@@ -0,0 +1,138 @@
+    friend class File;
+    friend class Directory;
+
+public:
+    // Note: StatsFileSystem takes ownerhip of the underlying oss
+    FileSystem(XrdOss *oss, XrdSysLogger *log, const char *configName, XrdOucEnv *envP);
+    virtual ~FileSystem();
+
+    bool
+    Config(const char *configfn);
+
+    // Indicate whether the initialization of the object was successful
+    //
+    // If `false` and `errMsg` is set, then the failure is fatal and the
+    // xrootd server should not startup.
+    // If `false` and `errMsg` is empty, then it's OK to bypass this object
+    // and just use the wrapped OSS pointer directly.
+    bool
+    InitSuccessful(std::string &errMsg);
+
+    XrdOssDF *newDir(const char *user=0) override;
+    XrdOssDF *newFile(const char *user=0) override;
+    int Chmod(const char * path, mode_t mode, XrdOucEnv *env=0) override;
+    int       Rename(const char *oPath, const char *nPath,
+                         XrdOucEnv  *oEnvP=0, XrdOucEnv *nEnvP=0) override;
+    int       Stat(const char *path, struct stat *buff,
+                       int opts=0, XrdOucEnv *env=0) override;
+    int       StatFS(const char *path, char *buff, int &blen,
+                         XrdOucEnv  *env=0) override;
+    int       StatLS(XrdOucEnv &env, const char *path,
+                         char *buff, int &blen) override;
+    int       StatPF(const char *path, struct stat *buff, int opts) override;
+    int       StatPF(const char *path, struct stat *buff) override;
+    int       StatVS(XrdOssVSInfo *vsP, const char *sname=0, int updt=0) override;
+    int       StatXA(const char *path, char *buff, int &blen,
+                         XrdOucEnv *env=0) override;
+    int       StatXP(const char *path, unsigned long long &attr,
+                         XrdOucEnv  *env=0) override;
+    int       Truncate(const char *path, unsigned long long fsize,
+                           XrdOucEnv *env=0) override;
+    int       Unlink(const char *path, int Opts=0, XrdOucEnv *env=0) override;
+
+private:
+    static void * AggregateBootstrap(void *instance);
+    void AggregateStats();
+
+    XrdXrootdGStream* m_gstream{nullptr};
+
+    // Indicates whether the class was able to initialize.
+    // On initialization failure, if there is no failure message
+    // set, then we assume it is OK to proceed with the wrapped OSS.
+    // If m_failure is set, then we assume the initialization failure
+    // was fatal and it's better to halt startup than proceed.
+    bool m_ready{false};
+    std::string m_failure;
+    std::unique_ptr<XrdOss> m_oss;
+    XrdOucEnv *m_env;
+    XrdSysError m_log;
+
+    class OpTimer {
+        public:
+            OpTimer(std::atomic<uint64_t> &op_count, std::atomic<uint64_t> &slow_op_count, std::atomic<uint64_t> &timing, std::atomic<uint64_t> &slow_timing, std::chrono::steady_clock::duration duration);
+            ~OpTimer();
+
+        private:
+            std::atomic<uint64_t> &m_op_count;
```

Comment:
We still have the std::atomic issue. Changing this is RAtomics is trivial and no other code changes are needed. What's the issue here?

Reply:
Well, this never was brought up previously which is why it was never changed.

Hadn't previously `RAtomic` before, seems like it'd fit this use case well.  Will refactor.

Reply:
Refactoring done in the latest commit.